src/index.ts

```diff
@@ -19,6 +19,74 @@
  * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-export function helloWorld() {
-  return "hello world!";
-}
+// The only import we need from the Node AuthN library is the Session class.
+import { Session } from "@inrupt/solid-client-authn-node";
+import { InMemoryStorage } from "@inrupt/solid-client-authn-core";
+
+const argv = require("yargs/yargs")(process.argv.slice(2))
+  .describe(
+    "oidcIssuer",
+    "The identity provider at which the user should authenticate."
+  )
+  .alias("issuer", "oidcIssuer")
+  .describe("clientName", "The name of the bootstrapped app.")
+  .demandOption(["oidcIssuer"])
+  .locale("en")
+  .help().argv;
+
+import express from "express";
+const app = express();
+const PORT = 3001;
+const iriBase = `http://localhost:${PORT}`;
+const storage = new InMemoryStorage();
+
+// Initialised when the server comes up and is running...
+let session: Session;
+
+const server = app.listen(PORT, async () => {
+  session = new Session({
+    insecureStorage: storage,
+    secureStorage: storage,
+  });
+
+  console.log(`Listening at: [http://localhost:${PORT}].`);
+  console.log(`Logging in ${argv.oidcIssuer} to get a refresh token.`);
+  session
+    .login({
+      clientName: argv.clientName,
+      oidcIssuer: argv.oidcIssuer,
+      redirectUrl: iriBase,
+      tokenType: "DPoP",
+      handleRedirect: (url) => {
+        console.log(`\nPlease visit ${url} in a web browser.\n`);
+      },
+    })
+    .catch((e) => {
+      throw new Error(`Login failed: ${e.toString()}`);
+    });
+});
+
+app.get("/", async (_req, res) => {
+  console.log("Login successful.");
+  await session.handleIncomingRedirect(`${iriBase}${_req.url}`);
```

Comment:
I know it's kinda mandated by the common ESLint rules, but I'm just going to rage about it anyway: I don't think at all that `${iriBase}${_req.url}` is more easily readable than `iriBase + _req.url` :P